rmw_dds_common/test/test_time_utils.cpp

```diff
@@ -32,23 +32,53 @@ TEST(test_time_utils, test_unmodified_zeros)
 
 TEST(test_time_utils, test_unmodified_max)
 {
-  const rmw_time_t max_dds_time {0x7FFFFFFF, 0xFFFFFFFF};
+  // Maximum time length supported by DDS is { INT_MAX, 10^9 - 1 }
+  const rmw_time_t max_dds_time {0x7FFFFFFF, 999999999ULL};
   auto t2 = rmw_dds_common::clamp_rmw_time_to_dds_time(max_dds_time);
   EXPECT_TRUE(t2 == max_dds_time);
 }
 
 TEST(test_time_utils, test_seconds_overflow)
 {
-  const rmw_time_t slightly_too_large {2147483651, 0};
-  auto t3 = rmw_dds_common::clamp_rmw_time_to_dds_time(slightly_too_large);
-  const rmw_time_t result3 {0x7FFFFFFF, 4000000000};
-  EXPECT_TRUE(t3 == result3);
+  // Maximum time length supported by DDS is { INT_MAX, 10^9 - 1 }
+  const rmw_time_t slightly_too_large {0x80000000, 0};
+  auto t3_sec = rmw_dds_common::clamp_rmw_time_to_dds_time(slightly_too_large);
+  const rmw_time_t result3 {0x7FFFFFFF, 999999999ULL};
+  EXPECT_TRUE(t3_sec == result3);
+  // Set the whole length via the nanosec field.
+  const rmw_time_t slightly_too_large_ns {0, 0x80000000 * 1000000000ULL};
+  auto t3_ns = rmw_dds_common::clamp_rmw_time_to_dds_time(slightly_too_large_ns);
+  EXPECT_TRUE(t3_ns == result3);
```

Comment:
@asorbini consider adding a test case for when both seconds and nanoseconds are slightly above the limit.

Reply:
Test added in [1fb9751](https://github.com/ros2/rmw_dds_common/pull/48/commits/1fb97517ff4ed002e510664b6906deae39c4322f).